postgresql-11.5/src/bin/psql/commandbteq.c

```diff
@@ -345,43 +347,29 @@ static dotResult
 exec_command_set(BteqScanState scan_state, bool active_branch)
 {
     bool        success = true;
-
+    char        quote;
     if (active_branch)
     {
-        char       *opt0 = bteq_scan_dot_option(scan_state,
-                                                  OT_BTEQ_NORMAL, NULL, false);
-
-        if (!opt0)
+        char *opt0 = read_connect_arg(scan_state);
+        if (opt0)
         {
-            /* list all variables */
-            PrintVariables(pset.vars);
-            success = true;
-        }
-        else
-        {
-            /*
-             * Set variable to the concatenation of the arguments.
-             */
-            char       *newval;
             char       *opt;
-
-            opt = bteq_scan_dot_option(scan_state,
-                                         OT_BTEQ_NORMAL, NULL, false);
-            newval = pg_strdup(opt ? opt : "");
-            free(opt);
-
-            while ((opt = bteq_scan_dot_option(scan_state,
-                                                 OT_BTEQ_NORMAL, NULL, false)))
-            {
-                newval = pg_realloc(newval, strlen(newval) + strlen(opt) + 1);
-                strcat(newval, opt);
-                free(opt);
+            char       *ptr;
+            opt = read_connect_arg(scan_state);
+            if (opt && strcasecmp(opt0, "width") == 0) {
```

Comment:
instead of `if (opt)` write `if (opt != NULL)`.
And everywhere use explicit NULL check.